### ISSUE TITLE
upytercon 2020: JupyterHub ≠ x, for all x: A tale of two JupyterHubs

### ISSUE THREAD

**Issue by rkdarst** [jupyterhub/team-compass#337]

**Comment by rkdarst** [comment]
[I pushed enter while trying to edit the title of this, which submitted an issue with no content... sorry about that]

So, in #312, we talked about JupyterCon proposals.  My proposal got accepted (title in the issue title), and I'd like to continue discussing my topic with the team.

I have my initial outline here.  I welcome anyone to comment on it:
https://docs.google.com/document/d/11nZpTvzbReqz62zcxAIuEsbrZouNV01xFMjuWoq_mfM/edit

In addition to any advice with the outline, the things I would specifically welcome help with:
- What are the most common misconceptions you have heard about JupyterHub?  I can address these in the talk.
- What are some common configuration requests, which I could mention.  The section is currently titled "opinionated solutions to common problems".  Ideally, the opinion would match the JupyterHub team, not me - this section is kind of empty now and needs some content.
- Anything else you would like to address?  I think I'll add some mentions of binderhub as counterpoint to the very state-heavy instances I am currently talking about.
- Does anyone else want to become a co-author and present with me?  (I will also credit other contributors)

**Comment by rcthomas** [comment]
On your 3rd point, how do you feel about briefly addressing or contrasting with alternatives to JupyterHub?  It's JupyterCon and all of course, but I mention that because I wonder if touching on those alternatives might highlight needs that currently are going unmet by the JupyterHub project, and these unmet needs of course may represent opportunities for JupyterHub to develop further.

**Comment by manics** [comment]
> What are some common configuration requests, which I could mention

I've seen several issues recently on how to setup JupyterHub for cross-domain use, though I'm not sure if that's only become noticeable because the new `SameSite` cookie setting is now active in Chrome. See e.g.
- https://github.com/jupyterhub/jupyterhub/issues/3162
- https://discourse.jupyter.org/t/issue-with-chrome-80-and-samesite-cookie-changes/3757

**Comment by rkdarst** [comment]
> On your 3rd point, how do you feel about briefly addressing or contrasting with alternatives to JupyterHub?

Good point, we should definitely do this!  Unfortunately I don't know that much about alternatives - well, not enough to write a consistent explanation of them.

> I've seen several issues recently on how to setup JupyterHub for cross-domain use

... I'm thinking this may be a bit too technical, or topical right now, to be worth mentioning.  Though I'm thinking how I can turn it into some other lesson about how the different JH abstraction layers interact... (suggestions about this?)

Thanks for the feedback!

**Comment by manics** [comment]
I guess it comes under the more general topic of integrating JupyterHub with other resources. You've mentioned `Integrate with X, like LTI` in the doc. Is that one way (LTI for JupyterHub auth?) or two way (JupyterHub can interact with other LTI services?).

**Comment by rcthomas** [comment]
In terms of alternatives, Open OnDemand (https://openondemand.org/) often comes up as a way for HPC centers to provide Jupyter notebooks through an interface that also offers HPC app UI, RStudio, Matlab, etc.

**Comment by rkdarst** [comment]
I realized it would be good to have some slides or visual aid, so that slowed me down.  Also overcoming the fear of doing it.  Anyway, I've uploaded the video here for everyone to see: https://drive.google.com/file/d/1JcDqSBTmSPWdTDIDoDt9A9I7gioJ875x/view
@sgibson91 @rcthomas, I guess this is relevant to your interest.  I still have time to repeat the recording before today ends somewhere on earth.

I did this as a one-short recording (I was originally going to do it in parts and splice myself, but this seemed to work).  My overall evalution is OK, could be done again.  I was a lot faster than expected in the beginning, and slower at the end.  Overall the time was good.  The early part seems a bit too scripted and monotonous.  What do you all think?

**Comment by manics** [comment]
I like the intro, you've clearly said how JupyterHub is different from notebook and lab. Probably too late now, but if you get a chance I think it'd be worth explaining what Spawners are in more practical terms beyond being "resource managers", e.g. "they allocate resources and run the singleuser server...".

**Comment by yuvipanda** [comment]
\o/ Thanks for the talk :D
